velox/expression/CastExpr.cpp

```diff
@@ -420,6 +423,43 @@ VectorPtr CastExpr::applyRow(
       std::move(newChildren));
 }
 
+template <bool castTo>
+void applyCustomTypeCast(
+    const CastOperatorPtr& castOperator,
+    const SelectivityVector& nonNullRows,
+    VectorPtr& input,
+    exec::EvalCtx* context,
+    const TypePtr& otherType,
+    VectorPtr* result) {
+  LocalDecodedVector decoded(context, *input, nonNullRows);
+  auto inputDecoded = decoded.get();
+
+  exec::LocalSelectivityVector baseRows(
+      context->execCtx(), inputDecoded->base()->size());
+  baseRows->clearAll();
+  nonNullRows.applyToSelected(
+      [&](auto row) { baseRows->setValid(inputDecoded->index(row), true); });
+  baseRows->updateBounds();
+
+  VectorPtr localResult;
+  BaseVector::ensureWritable(
+      *baseRows, VARCHAR(), context->pool(), &localResult);
```

Comment:
VARCHAR() is not the right type here. Please, double check.

Reply:
Thank you for catching this! This was used when this piece of code was in JsonCastOperator before refactoring. I changed it to `thisType` that represents the custom type in the current version to support general cases.